### ISSUE TITLE
Dependencies inside Slash Command Jobs

### ISSUE THREAD

**Issue by richdynamix** [spatie/laravel-slack-slash-command#5]
First of all, been playing with this all day and its saved me hours of development. Big fan already. 👍 

I was looking at ways to change the format of the output as it can become hard to read if there are many items in the response. One way I was thinking was using the Jobs to break up the items into mx 5 responses, giving the output a break in between the lines.

Following on from your tutorial it mentions that you can inject the dependencies into the handle method of the job i.e. - 

```
namespace App\SlashCommand\Jobs;

use Spatie\SlashCommand\Jobs\SlashCommandResponseJob;

class SlowApiJobJob extends SlashCommandResponseJob
{
    // notice here that Laravel will automatically inject dependencies here
    public function handle(MyApi $myApi)
    {
        $response = $myApi->fetchResponse();

        $this
           ->respondToSlack("Here is your response: {$response}")
           ->send();
    }
}
```

However Laravel fails and complains that `Declaration of MyApp\Jobs\LatestPostsJob::handle(MyApp\Contracts\PostRepository $postRepository) must be compatible with Spatie\SlashCommand\Jobs\SlashCommandResponseJob::handle()`

I tried to add this to a construct method but I got the following - `Symfony\Component\Debug\Exception\FatalThrowableError: Type error: Argument 1 passed to MyApp\Jobs\LatestPostsJob::__construct() must be an instance of MyApp\Contracts\PostRepository, none given`

Is there something I am missing?

**Comment by freekmurze** [comment]
I'm thinking that I shouldn't have coded the `handle` method into the abstract base class. I'll investigate soon and see if the package needs to be updated.

In the meantime you could try to to remove `handle` from `SlashCommandResponseJob` (in the vendor folder) to see if that works for you.

**Comment by richdynamix** [comment]
Yeah dependencies work fine when I remove the abstract handle method. Relating to the original question of formatting. Is it possible to add a break point in the output. Another words break the solid green line for better, clearer display of multiple items?

**Comment by freekmurze** [comment]
Mmmm, I do not understand your question. Are you talking about the formatting of the code in the examples in the docs?

Could you show me some code to make your question clear?

**Comment by richdynamix** [comment]
Let me start a new issue so we don't pollute this thread.

**Comment by freekmurze** [comment]
`SlashCommandResponseJob` is fixed in the newly tagged release `1.0.1`.
